core/src/main/java/com/linecorp/armeria/server/ServiceRequestContext.java

```diff
@@ -61,33 +60,54 @@
 
     /**
      * Returns the server-side context of the {@link Request} that is being handled in the current thread.
+     * If the context is a {@link ClientRequestContext}, {@link ClientRequestContext#root()} is returned.
      *
      * @throws IllegalStateException if the context is unavailable in the current thread or
-     *                               the current context is not a {@link ServiceRequestContext}.
+     *                               the current context is a {@link ClientRequestContext} and
+     *                               {@link ClientRequestContext#root()} is {@code null}
      */
     static ServiceRequestContext current() {
         final RequestContext ctx = RequestContext.current();
-        checkState(ctx instanceof ServiceRequestContext,
-                   "The current context is not a server-side context: %s", ctx);
-        return (ServiceRequestContext) ctx;
+        if (ctx instanceof ServiceRequestContext) {
+            return (ServiceRequestContext) ctx;
+        }
+
+        final ServiceRequestContext root = ((ClientRequestContext) ctx).root();
+        if (root != null) {
+            return root;
+        }
+
+        throw new IllegalStateException(
+                "The current context is not a server-side context and does not have a root " +
```

Comment:
nit: The Javadoc said `the current context is a {@link ClientRequestContext} and`. Don't we need to synchronize this too or revert the Javadoc?

Reply:
I synced it. 😄

Reply:
Well let me revert this, because the context is not a client request context as well. It could be a fakeContext in a unit test.

Reply:
In that case, the Javadoc still makes sense I guess.

Reply:
Ah, I misunderstood. Let revert this. :-)